include/deal.II/base/mpi_consensus_algorithm.h

```diff
@@ -0,0 +1,206 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_mpi_consensus_algorithm_h
+#define dealii_mpi_consensus_algorithm_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/mpi.h>
+#include <deal.II/base/mpi.templates.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace Utilities
+{
+  namespace MPI
+  {
+    /**
+     * This class implements Utilities::MPI::ConsensusAlgorithmProcess,
+     * using user-provided function wrappers.
+     * The advantage of this class is that users do not have to write their
+     * own implementation but can register lambda functions directly.
+     */
+    template <typename T1, typename T2>
+    class AnonymousConsensusAlgorithmProcess
+      : public ConsensusAlgorithmProcess<T1, T2>
+    {
+    public:
+      /**
+       * Register functions that should be called when calling the methods of
+       * ConsensusAlgorithmProcess.
+       *
+       * @param function_compute_targets called during `compute_targets`.
+       * @param function_create_request called during `create_request`.
+       * @param function_answer_request called during `answer_request`.
+       * @param function_prepare_buffer_for_answer called during
+       *   `prepare_buffer_for_answer`.
+       * @param function_read_answer called during `read_answer`.
+       */
+      AnonymousConsensusAlgorithmProcess(
+        const std::function<std::vector<unsigned int>()>
+          &function_compute_targets,
+        const std::function<void(const unsigned int, std::vector<T1> &)>
+          &function_create_request =
+            [](const unsigned int, std::vector<T1> &) {},
```

Comment:
I guess you prefer not to check for valid non-empty functions in the implementation below - otherwise you could have simply written `{}` as default argument. I am fine with the current setup, as it plays the ball back to the user in case they get a bad function exception.